Activate GitHub workflow for online building of SCInsta IPA

zeuikli opened the pull request:
Example : https://github.com/zeuikli/BHInsta/actions/runs/8732321318

Run workflow : 
![CleanShot 2024-04-18 at 12 10 19@2x](https://github.com/SoCuul/BHInsta/assets/7897668/3f5b105c-002f-460a-9f1d-34facbd76328)


Please enter the iOS SDK version and the IPA URL, then wait for the completion of the IPA build.

Build Status : ![CleanShot 2024-04-18 at 12 23 38@2x](https://github.com/SoCuul/BHInsta/assets/7897668/703facaf-bb78-4f95-9e25-eb061b493d4c)

Confirm that the IPA file can be installed via eSign, but note that the image download feature is still not functioning properly.

SoCuul commented on .github/workflows/buildapp.yml:
```diff
@@ -0,0 +1,189 @@
+      - name: Download iOS SDK
+        if: steps.SDK.outputs.cache-hit != 'true'
+        run: |
+          git clone --quiet -n --depth=1 --filter=tree:0 https://github.com/arichornlover/sdks/
```
If possible it'd be great to download the SDKs from https://github.com/xybp888/iOS-SDKs

SoCuul commented on .github/workflows/buildapp.yml:
```diff
@@ -0,0 +1,189 @@
+          Instagram_URL: ${{ inputs.decrypted_instagram_url }}
+          
+      - name: Build BHInsta tweak for sideloading (as IPA)
+        run: |
```
If this could be made to run the build script instead that would be great.
Having duplicated building code could cause confusions in the future.

SoCuul commented on .github/workflows/buildapp.yml:
```diff
@@ -0,0 +1,189 @@
+# Original idea by @ISnackable. Thanks to him for handling the hardest parts!
```
Maybe credit the uYouEnhanced script up here in a comment as well, since they made this base script haha.

zeuikli commented:
After the review, you can merge your commit into the main branch.

zeuikli commented:
Update build.sh to fix the issue of IPA file not being found.

Added script commands to install pyzule.
#Install pyzule
bash -c "$(curl https://raw.githubusercontent.com/asdfzxcvbn/pyzule/main/install-pyzule.sh)"

Modified the build.sh script by replacing certain instructions originally designated for BHInsta with those intended for SCInsta.

You can review my latest PR to verify these changes.

SoCuul commented:
> Update build.sh to fix the issue of IPA file not being found.
> 
> Added script commands to install pyzule. #Install pyzule bash -c "$(curl https://raw.githubusercontent.com/asdfzxcvbn/pyzule/main/install-pyzule.sh)"
> 
> Modified the build.sh script by replacing certain instructions originally designated for BHInsta with those intended for SCInsta.
> 
> You can review my latest PR to verify these changes.

It's super inefficient to install pyzule each time someone tries to build it, as they only need to install it once.

Could you just add it as another step in the buildapp workflow?

Also, just leave build.sh as it is in the main branch for now, as I want to keep the functionality the same atm.

SoCuul reviewed (changes requested):
All that needs to be changed is moving the pyzule stuff to a build step and reverting the changes in `build.sh`. The script should run fine if in the workflow step it cds into the folder, as all the paths in the shell script are relative to the current directory.

zeuikli commented:
Initially, you can proceed with the merge, but it's imperative to ensure that the "pyzule" tool is installed using GitHub Actions to facilitate the building of the IPA.

SoCuul commented:
> Initially, you can proceed with the merge, but it's imperative to ensure that the "pyzule" tool is installed using GitHub Actions to facilitate the building of the IPA.

I do not feel comfortable merging this into the main branch in its current state, as it would cause confusion for users who do not understand why the workflow fails on their end.

SoCuul commented:
@zeuikli If you're working on this, what do you think about adding a dropdown in the workflow settings to select between sideloaded ipa, rootless and rootful

zeuikli commented:
@SoCuul I have already updated the Github workflow and package dependencies to enable building IPA and deb files.
You can review my pull request, but since your settings are for local builds, I recommend not merging my PR into the main branch. Instead, you can create another branch for GitHub Actions use cases.

SoCuul commented:
> @SoCuul I have already updated the Github workflow and package dependencies to enable building IPA and deb files. You can review my pull request, but since your settings are for local builds, I recommend not merging my PR into the main branch. Instead, you can create another branch for GitHub Actions use cases.

Thank you so much for your work! I'll take a look at the PR when I can.

ic3dlem0n commented:
W zeuiki now sideloading this won't more complicated than it already is

SoCuul commented:
I still plan on implementing this when I can find the time. Kinda burned out rn from working on this atm.

SoCuul commented:
@zeuikli Thank you so much for your work on this! Incredibly appreciated by me 😄
